app/models/dialog.rb

```diff
@@ -66,6 +66,14 @@ def validate_children
       errors.add(:base, _("Dialog field name cannot be duplicated on a dialog: %{duplicates}") % {:duplicates => duplicate_field_names.join(', ')})
     end
 
+    associations_on_dialog_fields = DialogFieldAssociation.select { |dfa| dialog_fields.pluck(:id).include?(dfa.trigger_id) }
+    if associations_on_dialog_fields
+      associations = {}
+      associations_on_dialog_fields.each { |dfa| associations.merge!(DialogField.find(dfa.trigger_id).name => [DialogField.find(dfa.respond_id).name]) }
```

Comment:
We shouldn't have to do a `DialogField.find` here right? We have `dialog_fields` as a relationship so that should do the lookup once and then cache it so we don't have to hit the database two extra times here.